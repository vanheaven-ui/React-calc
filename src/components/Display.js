import PropTypes from 'prop-types';

const Display = ({ output, nxt, opr }) => (
  <div className="display-window">
    {output}
    {opr}
    {nxt}
  </div>
);

Display.defaultProps = {
  output: '0',
  nxt: '',
  opr: '',
};

Display.propTypes = {
  output: PropTypes.string,
  nxt: PropTypes.string,
  opr: PropTypes.string,
};

export default Display;
