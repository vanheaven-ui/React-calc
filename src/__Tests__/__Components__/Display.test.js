import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Display from '../../components/Display';
import '@testing-library/jest-dom';

test('is rendered without crashing', () => {
  const rootNode = document.createElement('div');
  ReactDOM.render(<Display />, rootNode);
});

test('doesnot change unexpectedly', () => {
  const displayComp = renderer.create(<Display />);
  let tree = displayComp.toJSON();
  expect(tree).toMatchSnapshot();
});
