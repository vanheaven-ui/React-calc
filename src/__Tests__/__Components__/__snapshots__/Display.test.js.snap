// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`doesnot change unexpectedly 1`] = `
<div
  className="display-window"
>
  0
  
  
</div>
`;
